pkg/ebpf/benchmark_test.go

```diff
@@ -0,0 +1,156 @@
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os/exec"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	bpflib "github.com/DataDog/ebpf"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	bpfToolPath       string
+	currKernelVersion uint32
+	readVars          sync.Once
+)
+
+type bpftoolStats struct {
+	RunTimeNS float64 `json:"run_time_ns"`
+	RunCnt    float64 `json:"run_cnt"`
+
+	XlatedBytes float64 `json:"bytes_xlated"`
+	JITedBytes  float64 `json:"bytes_jited"`
+}
+
+func readGlobalVars() {
+	var err error
+	currKernelVersion, err = bpflib.CurrentKernelVersion()
+	if err != nil {
+		log.Fatalf("unable to read kernel version: %s\n", err)
+	}
+	bpfToolPath, err = exec.LookPath("bpftool")
+	if err != nil {
+		log.Fatalf("unable to find bpftool in PATH: %s\n", err)
+	}
+}
+
+func setBPFStatsCollection(b *testing.B, enable bool) error {
+	if currKernelVersion < stringToKernelCode("5.1.0") {
+		b.Skip("kernel version must be at least 5.1.0 for kernel-level eBPF stats collection")
+		return nil
+	}
+
+	flag := 0
+	if enable {
+		flag = 1
+	}
+	cmd := exec.Command("sysctl", "-w", fmt.Sprintf("kernel.bpf_stats_enabled=%d", flag))
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("error enabling bpf stats: %s %w", out, err)
+	}
+	return nil
+}
+
+func collectBPFToolStats(progID uint32) (*bpftoolStats, error) {
+	cmd := exec.Command(bpfToolPath, "--json", "prog", "show", "id", fmt.Sprintf("%d", progID))
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("error collecting bpf stats: %s %w", out, err)
+	}
+	stats := bpftoolStats{}
+	err = json.Unmarshal(out, &stats)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling json from bpftool: %s", err)
+	}
+	return &stats, nil
+}
+
+func getAllProbeStats(t *Tracer) (map[string]*bpftoolStats, error) {
+	ids, err := t.getProbeProgramIDs()
+	if err != nil {
+		return nil, err
+	}
+
+	allStats := make(map[string]*bpftoolStats, len(ids))
+	for name, id := range ids {
+		stats, err := collectBPFToolStats(id)
+		if err != nil {
+			return nil, err
+		}
+		allStats[name] = stats
+	}
+	return allStats, nil
+}
+
+func RunEBPFBenchmark(b *testing.B, fn func(*testing.B)) {
```

Comment:
is this named this way to avoid it getting run automatically?  Or did I miss something and benchmarks/tests can also be postfixed with "Benchmark"?

Reply:
This is not a benchmark itself. It is the wrapper to run a benchmark func (`fn` argument) and collect ebpf probe timings. See https://github.com/DataDog/datadog-agent/pull/6230/files#diff-67a3fb8a685d370edc54a5f443fc70fdR14 for an actual benchmark.

Reply:
Ah gotcha, hadn't gotten that far yet ;). Thanks!